File: src/CHTL/CHTLNode/OriginNode.cpp
#include "OriginNode.h"
#include <sstream>

namespace CHTL {

OriginNode::OriginNode(const std::string& name)
    : BaseNode(name), originType_("html") {
}

void OriginNode::setOriginContent(const std::string& content) {
    originContent_ = content;
}

void OriginNode::setOriginType(const std::string& type) {
    originType_ = type;
}

std::string OriginNode::generateHTML() const {
    if (originType_ == "html") {
        return originContent_;
    }
    return "";
}

std::string OriginNode::generateCSS() const {
    if (originType_ == "css") {
        return originContent_;
    }
    return "";
}

std::string OriginNode::generateJS() const {
    if (originType_ == "javascript") {
        return originContent_;
    }
    return "";
}

} // namespace CHTL

File: src/CHTL/CHTLNode/OriginNode.java
package CHTL.CHTLNode;

/**
 * Represents an [Origin] block in the AST.
 * It contains raw content that should be passed through the generator
 * without modification.
 */
public class OriginNode implements BaseNode {
    private final String type; // e.g., Html, Style, JavaScript
    private final String content;

    public OriginNode(String type, String content) {
        this.type = type;
        this.content = content;
    }

    public String getType() {
        return type;
    }

    public String getContent() {
        return content;
    }

    @Override
    public <R> R accept(Visitor<R> visitor) {
        return visitor.visitOriginNode(this);
    }

    @Override
    public BaseNode clone() {
        return new OriginNode(this.type, this.content);
    }
}
